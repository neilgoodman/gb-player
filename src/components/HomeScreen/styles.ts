import { StyleSheet } from 'react-native';
import styleGuide from '../../styleGuide';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: styleGuide.colors.secondaryBackground,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: styleGuide.colors.white,
  },
});
