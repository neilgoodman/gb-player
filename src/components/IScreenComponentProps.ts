import { NavigationRoute, NavigationScreenProp } from 'react-navigation';

export default interface IScreenComponentProps {
  navigation: NavigationScreenProp<NavigationRoute>;
};
