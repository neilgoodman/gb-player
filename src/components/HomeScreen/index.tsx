import React from 'react';
import { Button, Text, View } from 'react-native';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import styles from './styles';

export interface IProps {
  navigation: NavigationScreenProp<NavigationRoute>;
}

export default class HomeScreen extends React.Component<IProps, object> {
  public render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>hi hannah</Text>
      </View>
    );
  }
}
