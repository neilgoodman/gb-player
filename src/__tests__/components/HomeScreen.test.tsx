import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../../components/HomeScreen';
import IScreenComponentProps from '../../components/IScreenComponentProps';

describe('HomeScreen', () => {
  let props: IScreenComponentProps;

  beforeEach(() => {
    // @ts-ignore: jest mock
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
  });

  describe('render', () => {
    it('should render correctly when given all props', () => {
      const rendered = renderer
        .create(<HomeScreen navigation={props.navigation} />)
        .toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });
});
