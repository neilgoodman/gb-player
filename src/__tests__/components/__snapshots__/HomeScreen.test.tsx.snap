// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen render should render correctly when given all props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(44,46,48)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(255,255,255)",
      }
    }
  >
    Home Screen 
    
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          Array [
            Object {
              "backgroundColor": "rgb(81,84,88)",
              "color": "rgb(255,255,255)",
              "marginBottom": 40,
              "paddingBottom": 10,
              "paddingTop": 10,
              "width": 710,
            },
          ],
          null,
        ]
      }
    >
      Clear Auth Token
    </Text>
  </View>
</View>
`;
